packages/aws-lambda-graphql/src/Server.ts

```diff
@@ -227,16 +248,65 @@ export class Server<
         // based on routeKey, do actions
         switch (event.requestContext.routeKey) {
           case '$connect': {
+            const {
+              onWebsocketConnect,
+            } = this.subscriptionOptions || {};
+
             // register connection
             // if error is thrown during registration, connection is rejected
             // we can implement some sort of authorization here
             const endpoint = extractEndpointFromEvent(event);
 
-            await this.connectionManager.registerConnection({
+            const connection = await this.connectionManager.registerConnection({
               endpoint,
               connectionId: event.requestContext.connectionId,
             });
 
+            let newConnectionContext = {}
+
+            if (onWebsocketConnect) {
+              try {
+                const result = await onWebsocketConnect(
+                  connection,
+                  event,
+                  lambdaContext,
+                );
+
+                if (result === false) {
+                  throw new Error('Prohibited connection!');
+                } else if (result !== null && typeof result === 'object') {
+                  newConnectionContext = result
+                }
+              } catch (err) {
+                const errorResponse = formatMessage({
+                  type: SERVER_EVENT_TYPES.GQL_ERROR,
+                  payload: { message: err.message },
+                });
+
+                await this.connectionManager.sendToConnection(
+                  connection,
+                  errorResponse,
+                );
+                await this.connectionManager.closeConnection(connection);
```

Comment:
This one should be replaced with `this.connectionManager.unregisterConnecition` which will delete the connection from Connection store but won't call `ApiGatewayManager`.